### PR TITLE
RE2022-79: Add matchers endpoint for collections

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: src/service/data_product_specs.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 Note that "builtin" is a reserved ID for data products.
 """
 
+# NOTE: Once a collection has been saved with a data product, the data product cannot be
```
Comment (dauglyon) [inline src/service/data_product_specs.py:12]: This seems concerning....

Comment (dauglyon) [reply src/service/data_product_specs.py:12]: Any way to prevent this?

**Review by dauglyon — APPROVED** [review]
Approved with one noted concern

File: src/service/data_product_specs.py (hunk 1 above)
Comment (MrCreosote) [reply src/service/data_product_specs.py:12]: I don't think so. If the collection says it included data product A but the code to support data product A has been removed from the service, unexpected things are going to happen.

What would have to happen is that we'd need an updater script to go through the collections and remove the data product in question (and any matchers that might depend on the data product)

**Pull request merged**